services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/ExecutionEngineService.java

```diff
@@ -44,58 +45,95 @@ public ExecutionEngineService(
 
   public void initialize(EventChannels eventChannels) {
     eventChannels.subscribe(ChainHeadChannel.class, this);
-    eventChannels.subscribe(FinalizedCheckpointChannel.class, this);
```

Comment:
Won't the finality update be missed in the case when epoch processing happens with an empty slot and finalizes some of the previous checkpoints without the change of the head. I guess implementation of this fork choice update signal that addresses all corner cases could be a tricky one and we might be OK with the current solution for the Interop while leaving a technical debt here to implement production ready solution later on.

Reply:
Just checked that `FinalizedCheckpointChannel` event is also fired just on the head update (i.e. block import). It's not fired on empty slot event. 
So suggesting to leave that for now. If EE would really need that immediate finalization update, we will invent something